### ISSUE TITLE
IllegalMonitorStateException because of HazelcastInstanceNotActiveException

### ISSUE THREAD

**Issue by sravanith** [hazelcast/hazelcast#10467]
In our cluster we have 2 members where each has 2 Hazelcast Instances. And 2 jvm where each has 3 client connections. Using Hazelcast 3.3.1 . Issue occurs once in 20 days. This will get resolved if we restart clients.

```java
17-04-12 22:05:13|Apr 12, 2017 10:05:13 PM com.hazelcast.nio.tcp.SocketConnector
17-04-12 22:05:13|INFO: [#101]:5701 [app_#100] [3.3.1] Could not connect to: #100/10.183.130.90:5701. Reason: SocketException[Connection timed out: connect to address #100/10.183.130.90:5701]
17-04-12 22:05:13|Apr 12, 2017 10:05:13 PM com.hazelcast.nio.tcp.TcpIpConnectionMonitor
17-04-12 22:05:13|WARNING: [#101]:5701 [app_#100] [3.3.1] Removing connection to endpoint Address[#100]:5701 Cause => java.net.SocketException {Connection timed out: connect to address #100/10.183.130.90:5701}, Error-Count: 5
17-04-12 22:05:13|Apr 12, 2017 10:05:13 PM com.hazelcast.cluster.ClusterService
17-04-12 22:05:13|INFO: [#101]:5701 [app_#100] [3.3.1] Master Address[#100]:5701 left the cluster. Assigning new master Member [#101]:5701 this
17-04-12 22:05:13|Apr 12, 2017 10:05:13 PM com.hazelcast.cluster.ClusterService
17-04-12 22:05:13|INFO: [#101]:5701 [app_#100] [3.3.1] Removing Member [#100]:5701
17-04-12 22:05:13|Apr 12, 2017 10:05:13 PM com.hazelcast.cluster.ClusterService
17-04-12 22:05:13|INFO: [#101]:5701 [app_#100] [3.3.1] 
17-04-12 22:05:13|
17-04-12 22:05:13|Members [1] {
17-04-12 22:05:13|	Member [#101]:5701 this
17-04-12 22:05:13|}
17-04-12 22:05:13|
17-04-12 22:05:15|Apr 12, 2017 10:05:15 PM com.hazelcast.spi.OperationService
17-04-12 22:05:15|WARNING: [#101]:5701 [app_#100] [3.3.1] Member [#100]:5701 has left cluster!




17-04-12 22:07:19|Apr 12, 2017 10:07:19 PM com.hazelcast.partition.InternalPartitionService
17-04-12 22:07:19|WARNING: [#101]:5701 [app_#100] [3.3.1] This is the master node and received a PartitionRuntimeState from Address[#100]:5701. Ignoring incoming state! 
17-04-12 22:07:21|Apr 12, 2017 10:07:21 PM com.hazelcast.cluster.TcpIpJoiner
17-04-12 22:07:21|INFO: [#101]:5701 [app_#100] [3.3.1] Address[#101]:5701 is merging to Address[#100]:5701, because : joinRequest.getMemberCount() > currentMemberCount [2 > 1]

17-04-12 22:07:21|WARNING: [#101]:5701 [app_#100] [3.3.1] Preparing to merge... Waiting for merge instruction...

17-04-12 22:07:22|Apr 12, 2017 10:07:22 PM com.hazelcast.partition.InternalPartitionService
17-04-12 22:07:22|WARNING: [#101]:5701 [app_#100] [3.3.1] Owner of partition is being removed! Possible data loss for partition[270]. PartitionReplicaChangeEvent{partitionId=270, replicaIndex=0, oldAddress=Address[#101]:5701, newAddress=null}

17-04-12 22:07:22|Apr 12, 2017 10:07:22 PM com.hazelcast.partition.InternalPartitionService
17-04-12 22:07:22|INFO: [#101]:5701 [app_#100] [3.3.1] Initializing cluster partition table first arrangement...

17-04-12 22:07:22|Apr 12, 2017 10:07:22 PM com.hazelcast.concurrent.lock.operations.UnlockOperation
17-04-12 22:07:22|SEVERE: [#101]:5701 [app_#100] [3.3.1] Current thread is not owner of the lock! -> Owner: 54c86381-6a9c-4689-9eef-36260043eb94, thread-id: 1
17-04-12 22:07:22|java.lang.IllegalMonitorStateException: Current thread is not owner of the lock! -> Owner: 54c86381-6a9c-4689-9eef-36260043eb94, thread-id: 1
17-04-12 22:07:22|	at com.hazelcast.concurrent.lock.operations.UnlockOperation.ensureUnlocked(UnlockOperation.java:71)
17-04-12 22:07:22|	at com.hazelcast.concurrent.lock.operations.UnlockOperation.unlock(UnlockOperation.java:65)
17-04-12 22:07:22|	at com.hazelcast.concurrent.lock.operations.UnlockOperation.run(UnlockOperation.java:57)
17-04-12 22:07:22|	at com.hazelcast.spi.impl.BasicOperationService$OperationHandler.handle(BasicOperationService.java:699)
17-04-12 22:07:22|	at com.hazelcast.spi.impl.BasicOperationService$OperationHandler.access$400(BasicOperationService.java:675)
17-04-12 22:07:22|	at com.hazelcast.spi.impl.BasicOperationService.runOperationOnCallingThread(BasicOperationService.java:229)
17-04-12 22:07:22|	at com.hazelcast.spi.impl.BasicInvocation.doInvokeLocal(BasicInvocation.java:286)
17-04-12 22:07:22|	at com.hazelcast.spi.impl.BasicInvocation.doInvoke(BasicInvocation.java:272)
17-04-12 22:07:22|	at com.hazelcast.spi.impl.BasicInvocation.invoke(BasicInvocation.java:232)
17-04-12 22:07:22|	at com.hazelcast.spi.impl.BasicInvocationBuilder.invoke(BasicInvocationBuilder.java:45)
17-04-12 22:07:22|	at com.hazelcast.client.impl.client.PartitionClientRequest.process(PartitionClientRequest.java:60)
17-04-12 22:07:22|	at com.hazelcast.client.impl.ClientEngineImpl$ClientPacketProcessor.processRequest(ClientEngineImpl.java:439)
17-04-12 22:07:22|	at com.hazelcast.client.impl.ClientEngineImpl$ClientPacketProcessor.run(ClientEngineImpl.java:358)
17-04-12 22:07:22|	at com.hazelcast.spi.impl.BasicOperationService$BasicDispatcherImpl.dispatch(BasicOperationService.java:546)
17-04-12 22:07:22|	at com.hazelcast.spi.impl.BasicOperationScheduler$OperationThread.process(BasicOperationScheduler.java:439)
17-04-12 22:07:22|	at com.hazelcast.spi.impl.BasicOperationScheduler$OperationThread.doRun(BasicOperationScheduler.java:433)
17-04-12 22:07:22|	at com.hazelcast.spi.impl.BasicOperationScheduler$OperationThread.run(BasicOperationScheduler.java:408)
17-04-12 22:07:22|
17-04-12 22:07:22|Apr 12, 2017 10:07:22 PM com.hazelcast.nio.tcp.TcpIpConnection
17-04-12 22:07:22|INFO: [#101]:5701 [app_#100] [3.3.1] Connection [Address[10.183.130.92]:12801] lost. Reason: Socket explicitly closed
17-04-12 22:07:22|Apr 12, 2017 10:07:22 PM com.hazelcast.client.ClientEngine
17-04-12 22:07:22|SEVERE: [#101]:5701 [app_#100] [3.3.1] While executing request: com.hazelcast.queue.impl.client.PollRequest@6d74e3a6 -> Hazelcast instance is not ready yet!
17-04-12 22:07:22|com.hazelcast.core.HazelcastInstanceNotActiveException: Hazelcast instance is not ready yet!
17-04-12 22:07:22|	at com.hazelcast.client.impl.ClientEngineImpl$ClientPacketProcessor.processRequest(ClientEngineImpl.java:428)
17-04-12 22:07:22|	at com.hazelcast.client.impl.ClientEngineImpl$ClientPacketProcessor.run(ClientEngineImpl.java:358)
17-04-12 22:07:22|	at com.hazelcast.spi.impl.BasicOperationService$BasicDispatcherImpl.dispatch(BasicOperationService.java:546)
17-04-12 22:07:22|	at com.hazelcast.spi.impl.BasicOperationScheduler$OperationThread.process(BasicOperationScheduler.java:439)
17-04-12 22:07:22|	at com.hazelcast.spi.impl.BasicOperationScheduler$OperationThread.doRun(BasicOperationScheduler.java:433)
17-04-12 22:07:22|	at com.hazelcast.spi.impl.BasicOperationScheduler$OperationThread.run(BasicOperationScheduler.java:408)

17-04-12 22:07:28|INFO: [#101]:5701 [app_#100] [3.3.1] 
17-04-12 22:07:28|
17-04-12 22:07:28|Members [2] {
17-04-12 22:07:28|	Member [#100]:5701
17-04-12 22:07:28|	Member [#101]:5701 this
17-04-12 22:07:28|}
17-04-12 22:07:28|
```

Can you share info why this is happening.Thanks.

**Comment by jerrinot** [comment]
hello @sravanith,

Hazelcast 3.3 is 3 years old. Can you please try a newer version?

**Comment by jerrinot** [comment]
this looks like a duplicated of https://github.com/hazelcast/hazelcast/issues/8730

**Comment by sravanith** [comment]
@jerrinot 
Thanks for the response.
Is this because of the network issues or hazelcast issue?

**Comment by mdogan** [comment]
@sravanith: there are many fixes related to Lock in 3.5, 3.6 and 3.8. Please try a newer version. If problem still exists, feel free to reopen the issue.
